src/utility/matrixOps.cpp

```
@@ -2,49 +2,151 @@
+std::vector<float> matVecMul(const std::vector<std::vector<float>>& matrix, const std::vector<float>& vec) {
+    if(matrix.empty()) {
+        throw std::runtime_error("Matrix should not be empty");
+    }
+    if(vec.empty()) {
+        throw std::runtime_error("Vector should not be empty");
+    }
+
+    std::vector<float> res(matrix.size(), 0.0f);
+    for(size_t i = 0; i < matrix.size(); ++i) {
+        if(matrix[i].size() != vec.size()) {
+            throw std::runtime_error("All matrix rows dimensions need to match the vector size.");
```

Cppcheck (2.20.0)

[error] There is an unknown macro here somewhere. Configuration is required. If DEPTHAI_NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE is a macro then please configure it.

(unknownMacro)